### PR TITLE
Use URL-safe base64 encoding for auth

### PR DESCRIPTION
**Pull request by robin-wayve**
## What do these changes do?

Changes `compose_auth_header` to use `base64.urlsafe_b64encode` on the returned string. This aligns with [how the Docker CLI encodes the same header](https://github.com/docker/cli/blob/693ae6ca733193818020823129511f6be14eaae7/cli/command/registry.go#L30-L37).

## Are there changes in behavior for the user?

This allows authentication to private registries where the contents of the `auth` would require substitutions as described here: https://docs.python.org/3/library/base64.html#base64.urlsafe_b64encode

## Related issue number

Not from this project, but I figured out that I was running into the same issue described here: https://github.com/moby/moby/issues/41570

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `changes` folder
  * name it `<issue_id>.<type>` for example (588.bug)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -72,3 +72,8 @@ def test_clean_filters():
     filters = ()
     result = {"a": ["1", "2", "3", "4"], "b": ["string"]}
     utils.clean_filters(filters=filters) == json.dumps(result)
+
+
+def test_compose_auth_header():
+    auth = {"username": "alice", "password": "~"}
+    assert utils.compose_auth_header(auth) == "eyJ1c2VybmFtZSI6ICJhbGljZSIsICJwYXNzd29yZCI6ICJ-In0="
```
Comment (robin-wayve) [inline tests/test_utils.py:79]: I don't know if I should make this more obvious somehow, but the `-` would be a `+` if this wasn't `urlsafe_b64encode`.

**Comment by robin-wayve** [comment]
@asvetlov do you have capacity to review / merge this?

**Comment by robin-wayve** [comment]
This can be re-opened if desired.

**Comment by dwo** [comment]
@achimnol @bdraco @asvetlov any interest in getting this merged? I can re-do the PR if anyone would like to get this fix in.

**Comment by asvetlov** [comment]
Yes, please

**Pull request closed without merging**